### ISSUE TITLE
[Feedback] 'AmplifyRTNPushNotification-Swift.h' file not found

### ISSUE THREAD

**Issue by francesco-cerio** [aws-amplify/amplify-js#11246]
**Page**: [`/lib/push-notifications/getting-started/q/platform/[platform]`](https://docs.amplify.aws/lib/push-notifications/getting-started/q/platform/react-native)

**Feedback**:
Hi,
I've followed all the required steps to setup the React Native Push Notification client.
However, the iOS build fails with message:
**'AmplifyRTNPushNotification-Swift.h' file not found**

```
    "react": "18.2.0",
    "react-native": "0.71.5",
    "@aws-amplify/rtn-push-notification": "^1.1.0",
    "@react-native-async-storage/async-storage": "^1.18.1",
    "@react-native-community/netinfo": "^9.3.9",
    "amazon-cognito-identity-js": "^6.2.0",
    "aws-amplify": "^5.1.1",
    "react-native-get-random-values": "^1.8.0",
    "react-native-url-polyfill": "^1.3.0",
```

Thank you very much

**Comment by cshfang** [comment]
@francesco-cerio @minotogna Could you please share your Xcode version, full error log and what step you experienced this error at? The aforementioned file is supposed to be generated by Xcode.

**Comment by francesco-cerio** [comment]
Hi @cshfang,
my Xcode Version is 14.3 (14E222b).
The error I get is the string already mentioned before: 
../node_modules/@aws-amplify/rtn-push-notification/ios/AmplifyPushNotification.m:4:9 'AmplifyRTNPushNotification-Swift.h' file not found

I checked the PodFile and the minimum iOS target is 13.0 and I can successfully install pods.

**Comment by minotogna** [comment]
Hi @cshfang 

same Xcode version here.


Log is 
```
...
▸ Processing AmplifyRTNPushNotification-Info.plist
▸ Compiling AmplifyRTNPushNotification_vers.c
▸ Compiling AmplifyRTNPushNotification-dummy.m
▸ Compiling AmplifyRTNPushNotification.m
▸ Compiling AmplifyPushNotification.m

❌  /$MYPATH/node_modules/@aws-amplify/rtn-push-notification/ios/AmplifyPushNotification.m:4:9: 'AmplifyRTNPushNotification-Swift.h' file not found

#import "AmplifyRTNPushNotification-Swift.h"
                                        ^


▸ Building library libPermission-Notifications.a
▸ Building library libPermission-LocationWhenInUse.a
▸ Building library libPermission-LocationAlways.a
▸ Building library libPermission-LocationAccuracy.a
▸ Building library libPermission-Contacts.a
....
```

**Comment by cshfang** [comment]
Hmm ok. I appreciate your bearing with me on this but we need to figure out what permutation of setup will allow us to reproduce this on our end. Would you mind sharing with me some more information?
* Are you using RN CLI or Expo? (Seems like it's the former but I'd like to just verify)
* Could you share with me your project Podfile?
* Do you have the RN new architecture enabled?

**Comment by minotogna** [comment]
hi @cshfang 

I'm using RN CLI and new architecture is enabled. here the pod file

```
$RNFirebaseAsStaticFramework = true

require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '13.0'
prepare_react_native_project!
install! 'cocoapods', :deterministic_uuids => false

# Force pods to match minimum iOS version for React Native
# Fixes build issue on Xcode Cloud where some pods
# Use iOS 12 calls despite being set as iOS 11
def __apply_Xcode_14_3_RC_post_install_workaround(installer)
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      current_target = config.build_settings['IPHONEOS_DEPLOYMENT_TARGET']
      minimum_target = min_ios_version_supported
      if current_target.to_f < minimum_target.to_f
        config.build_settings['IPHONEOS_DEPLOYMENT_TARGET'] = minimum_target
      end
    end
  end
end

$RNFirebaseAnalyticsWithoutAdIdSupport = true


# If you are using a `react-native-flipper` your iOS build will fail when `NO_FLIPPER=1` is set.
# because `react-native-flipper` depends on (FlipperKit,...) that will be excluded
#
# To fix this you can also exclude `react-native-flipper` using a `react-native.config.js`
# ```js
# module.exports = {
#   dependencies: {
#     ...(process.env.NO_FLIPPER ? { 'react-native-flipper': { platforms: { ios: null } } } : {}),
# ```
#flipper_config = ENV['NO_FLIPPER'] == "1" ? FlipperConfiguration.disabled : FlipperConfiguration.enabled
flipper_config = FlipperConfiguration.disabled
#linkage = ENV['USE_FRAMEWORKS']
# if linkage != nil
#   Pod::UI.puts "Configuring Pod with #{linkage}ally linked Frameworks".green
#   use_frameworks! :linkage => linkage.to_sym
# end

use_frameworks! :linkage => :static

target 'MY_PROJECT' do
  $static_framework = []
  # React Native Maps dependencies
  rn_maps_path = '../node_modules/react-native-maps'

  config = use_native_modules!

  # Flags change depending on the env values.
  flags = get_default_flags()

  use_react_native!(
    :path => config[:reactNativePath],
    # Hermes is now enabled by default. Disable by setting this flag to false.
    # Upcoming versions of React Native may rely on get_default_flags(), but
    # we make it explicit here to aid in the React Native upgrade process.
    :hermes_enabled => flags[:hermes_enabled],
    :fabric_enabled => flags[:fabric_enabled],
    # Enables Flipper.
    #
    # Note that if you have use_frameworks! enabled, Flipper will not work and
    # you should disable the next line.
    :flipper_configuration => flipper_config,
    # An absolute path to your application root.
    :app_path => "#{Pod::Config.instance.installation_root}/.."
  )



  target 'MY_PROJECTTests' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable the next line.
  # use_flipper!()

  # MAGIC
  $static_framework += [
      'react-native-maps',
      'react-native-google-maps',
      'Google-Maps-iOS-Utils',
      'GoogleMaps',
      'RNPermissions',
      'Permission-LocationAccuracy',
      'Permission-LocationAlways',
      'Permission-LocationWhenInUse',
      'Permission-Notifications',
      'Permission-Contacts'
  ]

  pod 'react-native-google-maps', :path => rn_maps_path

  # ****** THIS IS THE MAGIC ******
  pre_install do |installer|
      Pod::Installer::Xcode::TargetValidator.send(:define_method, :verify_no_static_framework_transitive_dependencies) {}
          installer.pod_targets.each do |pod|
              if $static_framework.include?(pod.name)
                  def pod.build_type;
                  Pod::BuildType.static_library # >= 1.9
              end
          end
      end
  end

  post_install do |installer|
    react_native_post_install(
      installer,
      # Set `mac_catalyst_enabled` to `true` in order to apply patches
      # necessary for Mac Catalyst builds
      :mac_catalyst_enabled => false
    )
    __apply_Xcode_12_5_M1_post_install_workaround(installer)
    __apply_Xcode_14_3_RC_post_install_workaround(installer)
    # MAGIC
    installer.pods_project.targets.each do |target|
        if target.name == 'RCT-Folly'
          target.build_configurations.each do |config|
            config.build_settings['HEADER_SEARCH_PATHS'] = "$(inherited) ${PODS_ROOT}/fmt/include"
          end
        end
      end

    # Mixing Swift and Objective-C in a react-native project may be problematic.
    # Workaround:  https://github.com/facebookarchive/react-native-fbsdk/issues/755#issuecomment-787488994
    installer.aggregate_targets.first.user_project.native_targets.each do |target|
      target.build_configurations.each do |config|
        config.build_settings['LIBRARY_SEARCH_PATHS'] = ['$(inherited)', '$(SDKROOT)/usr/lib/swift']
      end
    end
    installer.pods_project.targets.each do |target|
      if target.respond_to?(:product_type) and target.product_type == "com.apple.product-type.bundle"
        target.build_configurations.each do |config|
            config.build_settings['CODE_SIGNING_ALLOWED'] = 'NO'
        end
      end
    end
  end

  pod 'GoogleSignIn', '~> 6.2.2'

  permissions_path = '../node_modules/react-native-permissions/ios'
  pod 'Permission-LocationAccuracy', :path => "#{permissions_path}/LocationAccuracy"
  pod 'Permission-LocationAlways', :path => "#{permissions_path}/LocationAlways"
  pod 'Permission-LocationWhenInUse', :path => "#{permissions_path}/LocationWhenInUse"
  pod 'Permission-Notifications', :path => "#{permissions_path}/Notifications"
  pod 'Permission-Contacts', :path => "#{permissions_path}/Contacts"
  pod 'RNFS', :path => '../node_modules/react-native-fs'

end
```

**Comment by tannerabread** [comment]
Hi @francesco-cerio, @minotogna this should be fixed following the merged PR above and the release of Amplify v5.1.2. Can you try it out and let us know if it works for you?

**Comment by tannerabread** [comment]
Hi 👋 Closing this as resolved. This should be handled with the recent version that came out, mentioned above.

If you are still experiencing this issue and in need of assistance, please feel free to comment and provide us with additional information so we can re-open this issue and be better able to assist you.

Thank you!
